src/db.rs

```diff
@@ -62,8 +65,13 @@ impl DieselPool {
         match self {
             DieselPool::Pool {
                 pool,
+                used_conns_metric,
                 time_to_obtain_connection_metric,
             } => time_to_obtain_connection_metric.observe_closure_duration(|| {
+                // Record the number of used connections before obtaining the current one.
+                let state = pool.state();
```

Comment:
I assume accessing `pool.state()` is an inexpensive operation?

Reply:
This just requires obtaining a mutex lock, so shouldn't be expensive.

Reply:
> requires obtaining a mutex lock

sounds pretty expensive to me 😅